### PR TITLE
Test

### PR DESCRIPTION
**Pull request by Rodin1000**

### CODE REVIEW

File: cgs_vmc/evaluation.py [hunk 1]
```diff
@@ -237,10 +237,10 @@ def run_evaluation(
           psi = session.run(wavefunction_value, feed_dict=feed_dict)
           config_ind = 0
           for i in range(0, batch_size):
-            out_file.write('({},{})\n'.format(psi[i].real, psi[i].imag))
+            out_file.write('{} {}\n'.format(psi[i].real, psi[i].imag))
```
Comment (kochkov92) [inline cgs_vmc/evaluation.py]: I think for the purpose of testing vectors with python we can just use pandas to read-in files. In this case we maintain uniformity with ED code, which is a nice tool to test and compare against. Here's a possible way to read the current format:
```
def get_wf(file_name):
  df = pd.read_csv(file_name)
  df = df.apply(lambda col: col.apply(lambda val: complex(val.strip('()'))))
  return df.values[:, 0]
```

File: cgs_vmc/evaluation.py [hunk 2]
```diff
@@ -237,10 +237,10 @@ def run_evaluation(
...
     config_batch = config_batch * 2. - 1.
     feed_dict = {placeholder_input: config_batch}
     psi = session.run(wavefunction_value, feed_dict=feed_dict)
     for i in range(0, config_ind):
-      out_file.write('({},{})\n'.format(psi[i].real, psi[i].imag))
+      out_file.write('{} {}\n'.format(psi[i].real, psi[i].imag))
```
Comment (kochkov92) [inline cgs_vmc/evaluation.py]: same as the above

File: cgs_vmc/normalizer.py [hunk 3]
```diff
@@ -47,3 +49,73 @@ def normalize_wavefunction(
   for _ in range(normalization_iterations):
     session.run(update_config)
     session.run(update_norm_on_batch)
+
```
Comment (kochkov92) [inline cgs_vmc/normalizer.py:52]: Since we are not deploying this part yet, I would move it to prototypes; then we can recover this part of the code when needed.

File: cgs_vmc/normalizer.py [hunk 4]
```diff
@@ -47,3 +49,73 @@ def normalize_wavefunction(
...
+  mc_step = graph_builders.get_monte_carlo_sampling(
+      shared_resources, configs, wavefunction)[0]
+  psi_value = wavefunction(configs)
+  max_value = tf.reduce_max(tf.square(psi_value))
+
+  return max_value, mc_step
+
+
+def run_normalization_ops(
+    max_value: Any,
```
Comment (kochkov92) [inline cgs_vmc/normalizer.py]: I think max_value is probably a np.float or something.

File: cgs_vmc/run_energy_evaluation.py [hunk 5]
```diff
@@ -25,6 +25,10 @@
     'checkpoint_dir', '',
     'Full path to the checkpoint directory.')
 
+flags.DEFINE_string(
+    'bond_file', '',
```
Comment (kochkov92) [inline cgs_vmc/run_energy_evaluation.py]: nit: `bonds_file`
Name of the file that contains bonds of the Hamiltonian.

File: cgs_vmc/run_supervised_training.py [hunk 6]
```diff
@@ -132,14 +133,20 @@ def main(argv):
     latest_checkpoint = tf.train.latest_checkpoint(hparams.checkpoint_dir)
     checkpoint_saver.restore(session, latest_checkpoint)
 
+  training_metrics_file = os.path.join(hparams.checkpoint_dir, \
```
Comment (kochkov92) [inline cgs_vmc/run_supervised_training.py]: do not use backslash line continuation except for `with` statement as per [link](http://google.github.io/styleguide/pyguide.html#32-line-length)

File: cgs_vmc/run_training.py [hunk 7]
```diff
@@ -21,6 +21,9 @@
 flags.DEFINE_string(
     'checkpoint_dir', '',
     'Full path to the checkpoint directory.')
+flags.DEFINE_string(
+    'bond_file', '',
```
Comment (kochkov92) [inline cgs_vmc/run_training.py]: Same as for run_supervised_training.py

File: cgs_vmc/training.py [hunk 8]
```diff
@@ -327,7 +337,7 @@ def build_opt_ops(
         shared_resources, configs, wavefunction)
 
     psi = wavefunction(configs)
-    psi_target = target_wavefunction(configs)
+    psi_target = target_wavefunction(configs) * hparams.scale
```
Comment (kochkov92) [inline cgs_vmc/training.py]: Since LogOverlapSWO doesn't depend on the norm mismatch there is no reason to scale `psi_target` here.

File: cgs_vmc/training.py [hunk 9]
```diff
@@ -459,7 +469,7 @@ def build_opt_ops(
     # )
 
     loss = tf.reduce_mean(
-        tf.squared_difference(psi, psi_target)
+        tf.squared_difference(psi, psi_target * hparams.scale)
```
Comment (kochkov92) [inline cgs_vmc/training.py]: I would move scaling up in the code to improve readability. One naturally would expect target to be target, not a target before scaling.

File: cgs_vmc/training.py [hunk 10]
```diff
@@ -254,7 +259,7 @@ def build_opt_ops(
     psi_target = target_wavefunction(configs)
 
     loss = tf.reduce_mean(
-        tf.squared_difference(psi, psi_target * np.sqrt(2 ** n_sites)))
+        tf.squared_difference(psi, psi_target * hparams.scale))
```
Comment (kochkov92) [inline cgs_vmc/training.py]: I think here you also want to move the the scaling up, so that `psi_target` has the scale you want.

File: cgs_vmc/training.py [hunk 11]
```diff
@@ -167,7 +167,7 @@ def build_opt_ops(
     psi_target = target_wavefunction(configs)
 
     loss = tf.reduce_mean(
-        tf.squared_difference(psi, psi_target * np.sqrt(2**n_sites)) /
+        tf.squared_difference(psi, psi_target * hparams.scale) /
```
Comment (kochkov92) [inline cgs_vmc/training.py]: I haven't thought about it before, but it makes more sense to put scale up to the `psi_target`, simply improves readability. (otherwise it's confusing that we are matching scaled target)

**Review by kochkov92 — CHANGES_REQUESTED** [review]
Thanks for fixing some typos. I think having `bonds_file` is a good idea. 
-) I suggest to stay compatible with the ED code on the wave function format at least for now.
-) I think rescaling at `psi_target` is more intuitive for someone who is reading the code.

**Comment by Rodin1000** [comment]
For normalizer.py, do you suggest to put the whole file into prototypes, or just keep the first function?
